polyply/src/gen_seq.py

```diff
@@ -53,14 +54,28 @@ def _random_replace_nodes(graph, residues, weights):
     graph: `:class:networkx.Graph`
     residues: list[str]
     weights: list[float]
+    attribute: str
+        the attribute to be replaced
+    seed: list[float]
+        a list of seeds to use in random selection
+        there needs to be one seed per node
```

Comment:
This is peculiar. Usually people would do the seeding themselves before invoking the program.

Reply:
problem is if you only set random.seed you will get always the same monomer/attribute being selected. This is rather for debugging than actual use in the sense that it allows you to get a distribution of monomers but you get the same every time. So in the test I can actually check that I get the resnames I would expect.